### ISSUE TITLE
False positive for generic.secrets.gitleaks.hashicorp-tf-password

### ISSUE THREAD

**Issue by Sjord** [returntocorp/sgrep-rules#3586]
[semgrep-rules/generic/secrets/gitleaks/hashicorp-tf-password.yaml at develop · semgrep/semgrep-rules](https://github.com/semgrep/semgrep-rules/blob/develop/generic/secrets/gitleaks/hashicorp-tf-password.yaml)

```
<input type=password autocomplete="new-password" />
```

```
$ semgrep -c 'r/generic.secrets.gitleaks.hashicorp-tf-password.hashicorp-tf-password' test.html

    test.html
     ❱ generic.secrets.gitleaks.hashicorp-tf-password.hashicorp-tf-password
          A gitleaks hashicorp-tf-password was detected which attempts to identify hard-coded credentials. It
          is not recommended to store credentials in source-code, as this risks secrets being leaked and used
          by either an internal or external malicious adversary. It is recommended to use environment        
          variables to securely provide credentials or retrieve credentials from a secure vault or HSM       
          (Hardware Security Module).                                                                        
          Details: https://sg.run/bw7lv                                                                      
                                                                                                             
            1┆ <input type=password autocomplete="new-password" />
```

**Comment by Sjord** [comment]
The upstream rule [gitleaks/cmd/generate/config/rules/hashicorp.go at master · gitleaks/gitleaks](https://github.com/gitleaks/gitleaks/blob/master/cmd/generate/config/rules/hashicorp.go) seems to limit the path of scanned files, maybe that's something that the semgrep rule can do as well:

```
Path:        regexp.MustCompile(`(?i)\.(?:tf|hcl)$`),
```

**Comment by LewisArdern** [comment]
Thanks @Sjord i'll need to look into it, gitleaks hasn't received updates in a while on our side sorry about that.

Going to take a look at it next week, and see if i can improve this also create a path include based on the variations in the regex' supplied for rules that have a path attached, since its an automated process

**Comment by LewisArdern** [comment]
For now I made a PR to fix it manually to fix the false positives, but will look to getting around to including latest rules and add the path include into the mix.
